### PR TITLE
Add support for immediate tiering

### PR DESCRIPTION
**Pull request by krisis**

### CODE REVIEW

**Review by poornas — APPROVED** [review]
LGTM

File: cmd/ilm/parse.go [hunk 1]
```diff
@@ -53,39 +53,22 @@ func extractILMTags(tagLabelVal string) []lifecycle.Tag {
...
 	}
 	if transitionDateSet && rule.Transition.StorageClass == "" {
-		return errors.New("if transitionDate or transitionDay is set, a valid storage class must be set")
-	}
-	if transitionDaySet && rule.Transition.StorageClass == "" {
-		return errors.New("if transitionDate or transitionDay is set, a valid storage class must be set")
-	}
-	if rule.Transition.StorageClass != "" && (!transitionDateSet && !transitionDaySet) {
-		return errors.New("if storage class is set, transitionDate or transitionDay must be set")
+		return errors.New("missing transitiion storage-class")
```
Comment (vadmeste) [inline cmd/ilm/parse.go:64]: s/transitiion/transition/

File: cmd/ilm/parse.go [hunk 2]
```diff
@@ -53,39 +53,22 @@ func extractILMTags(tagLabelVal string) []lifecycle.Tag {
...
 	}
 	return nil
 }
 
 func validateTranDays(rule lifecycle.Rule) error {
```
Comment (vadmeste) [inline cmd/ilm/parse.go:69]: Let's add when trans days is negative

File: go.mod [hunk 3]
```diff
@@ -61,3 +61,5 @@ require (
 	gopkg.in/yaml.v2 v2.4.0
 	maze.io/x/duration v0.0.0-20160924141736-faac084b6075
 )
+
+replace github.com/minio/minio-go/v7 => ../minio-go
```
Comment (vadmeste) [inline go.mod:65]: Not needed

**Review by vadmeste — COMMENTED** [review]
Some comments

**Review by vadmeste — APPROVED** [review]
LGTM

**Pull request merged**